src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -225,6 +229,30 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         }
     }
 
+    void ControlCore::Detach()
+    {
+        // Disable the renderer, so that it doesn't try to start any new frames
+        // for our engines while we're not attached to anything.
+        _renderer->WaitForPaintCompletionAndDisable(INFINITE);
+
+        _tsfTryRedrawCanvas.reset();
+        _updatePatternLocations.reset();
+        _updateScrollBar.reset();
+    }
+
+    void ControlCore::Reparent(const Microsoft::Terminal::Control::IKeyBindings& keyBindings)

```

Comment:
detach vs reparent seems like a verb category mismatch.

Also, why does it read like "reparent [this control to these] key bindings"